### PR TITLE
Do not automatically add primary key to attributes (or include[].attributes) if raw is set

### PR DESCRIPTION
**Pull request by pimterry**
This bring in the changes discussed in #3920, disabling the automatic addition of the primary key field to the attributes returned by a query, and thereby allowing aggregation in more cases, when raw is set to true.

This includes the changes and tests. Is there any documentation anywhere that needs to be updated in line with this?

### CODE REVIEW

**Comment by mickhansen** [comment]
Thank you for this PR. The only issue i have is passing `raw` as a single argument, perhaps we could pass the options instead?

File: test/integration/include/findAll.test.js [hunk 1]
```diff
@@ -1920,5 +1920,48 @@ describe(Support.getTestDialectTeaser('Include'), function() {
...
+            }
+          ],
+          raw: true
+        });
+      }).then(function (posts) {
+        expect(posts.length).to.equal(1);
+
+        var post = posts[0];
+
+        console.log("** " + JSON.stringify(post));
```
Comment (mickhansen) [inline test/integration/include/findAll.test.js]: Please remove the console.log

**Comment by pimterry** [comment]
Both those comments are now fixed.

Build has failed, but that looks to be for other unrelated reasons (issues with transaction serialization), which I suspect is something intermittent, but I don't have permission to retrigger the build. I can't see any way these couple of tiny markups could've caused that.

**Comment by mickhansen** [comment]
Thanks for the contribution :)

**Comment by hamza-sabri** [comment]
Sorry for opening this again. 
I read the article and I still couldn't find a way to do what the article suggested

I had three tables invoice, item, invoice_item(which had a quantity column)
I am doing a join between invoice and item models and would like to apply some aggregate functions (sum, min, max, count, avg, ... etc) on the result of the join or simply add up the quantity field from the result.


But I had the same error described here (https://www.softwire.com/insights/fixing-aggregation-over-relationships-with-sequelize/) and it was the closest I came to a solution but couldn't figure it out. 

could someone help?

**Pull request merged**